src/main/java/org/takes/rq/RqChunk.java

```diff
@@ -47,17 +47,10 @@
      */
     public RqChunk(final Request req) {
         super(
-            new Request() {
-                @Override
-                public Iterable<String> head() throws IOException {
-                    return req.head();
-                }
-
-                @Override
-                public InputStream body() throws IOException {
-                    return RqChunk.cap(req);
-                }
-            }
+            new RequestOf(
+                req::head,
+                () -> RqChunk.cap(req)
```

Comment:
@fanifieiev use the secondary construtor.

Reply:
@fabriciofx  I cannot use the secondary constructor here. See above.

Reply:
@fanifieiev as I said [here](https://github.com/yegor256/takes/pull/995#discussion_r347074093): *Of couse you can't use it in all places (e.g. beacause it can throws an Exception). So, in these places you can use the primary constructor...*